xarray/backends/api.py

```diff
@@ -415,6 +421,57 @@ def _dataset_from_backend_dataset(
     return ds
 
 
+def _datatree_from_backend_datatree(
+    backend_tree,
+    filename_or_obj,
+    engine,
+    chunks,
+    cache,
+    overwrite_encoded_chunks,
+    inline_array,
+    chunked_array_type,
+    from_array_kwargs,
+    **extra_tokens,
+):
+    if not isinstance(chunks, int | dict) and chunks not in {None, "auto"}:
+        raise ValueError(
+            f"chunks must be an int, dict, 'auto', or None. Instead found {chunks}."
+        )
+
+    _protect_datatree_variables_inplace(backend_tree, cache)
+    if chunks is None:
+        tree = backend_tree
+    else:
+        tree = DataTree.from_dict(
+            {
+                path: _chunk_ds(
+                    node.dataset,
+                    filename_or_obj,
+                    engine,
+                    chunks,
+                    overwrite_encoded_chunks,
+                    inline_array,
+                    chunked_array_type,
+                    from_array_kwargs,
+                    **extra_tokens,
+                )
+                for path, [node] in group_subtrees(backend_tree)
+            }
+        )
+
+        for path, [node] in group_subtrees(backend_tree):
+            tree[path].set_close(node._close)
+
+    # Ensure source filename always stored in dataset object
+    if "source" not in tree.encoding:
+        path = getattr(filename_or_obj, "path", filename_or_obj)
+
+        if isinstance(path, str | os.PathLike):
+            tree.encoding["source"] = _normalize_path(path)
```

Comment:
Do you want to add this encoding to all the nodes?

Reply:
I mean I'm not proposing adding it to each variable either, just to the group attributes.

FWIW this is a separable issue from supporting `chunks`, so can be punted to a future PR.

Reply:
I meant as an analogy that we don't do that, I'm not suggesting we'd have to put the encoding on the variables if we put them on the groups. My mental model is that this is a property of the object returned by the `open_*` function.

But either way I agree that we can punt that discussion to a future PR.